node/pkg/p2p/registry_test.go

```diff
@@ -0,0 +1,60 @@
+package p2p
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/mr-tron/base58"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestNewRegistry(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, "", registry.guardianAddress)
+	assert.Equal(t, 0, len(registry.errorCounters))
+	assert.Equal(t, 0, len(registry.networkStats))
+}
+
+func TestSetGuardianAddress(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, "", registry.guardianAddress)
+
+	registry.SetGuardianAddress("foo")
+	assert.Equal(t, "foo", registry.guardianAddress)
+}
+
+func TestSetNetworkStats(t *testing.T) {
+	registry := NewRegistry()
+	assert.Equal(t, 0, len(registry.networkStats))
```

Comment:
Don't test the same code twice. You covered that in `TestNewRegistry`.

```suggestion
```

Reply:
generally, this behavior I'm exhibiting here is coming from a mindset where parallel test execution is possible and state pollution could occur.  I also like the idea of seeing the state change in a give test, as a guarantee I'm getting the same starting point.  That said, I'll pull those, just noting why my thinking behind it.